packages/fuel-streams/src/data/logs/subjects.ts

```diff
@@ -0,0 +1,81 @@
+import type { BytesLike, Block as FuelBlock } from 'fuels';
+
+type BlockHeight = number;
+type Bytes32 = BytesLike;
+
+export class LogsSubject {
+  blockHeight?: BlockHeight;
+  txId?: Bytes32;
+  receiptIndex?: number;
+  logId?: Bytes32;
+
+  static readonly WILDCARD = 'logs.>';
+
+  constructor(
+    blockHeight?: BlockHeight,
+    txId?: Bytes32,
+    receiptIndex?: number,
+    logId?: Bytes32,
+  ) {
+    this.blockHeight = blockHeight;
+    this.txId = txId;
+    this.receiptIndex = receiptIndex;
+    this.logId = logId;
+  }
+
+  static wildcard(
+    blockHeight?: BlockHeight,
+    txId?: Bytes32,
+    receiptIndex?: number,
+    logId?: Bytes32,
+  ): string {
+    return new LogsSubject(
+      blockHeight,
+      txId,
+      receiptIndex,
+      logId,
+    ).parseWildcard();
+  }
+
+  parse(): string {
+    const parts = [
+      'logs',
+      this.blockHeight?.toString() || '*',
+      this.txId || '*',
+      this.receiptIndex?.toString() || '*',
+      this.logId || '*',
```

Comment:
These can be made dynamic like we have in Rust's, but this time at run-time

Reply:
@Jurshsmith can you elaborate more  ? I dont think I follow. ...